Tempo: Config and doc updates (#64017)
grafana/grafana: docs/sources/datasources/tempo/_index.md

> You can also configure settings specific to the Tempo data source:

Reviewer: Changing this to a period since this sentence isn't followed by a bulleted or numbered list. We could add a bulleted item for each of the subheadings that explain how to configure settings (there are a lot) or just add a note that the options are described below. For example: 
```suggestion
You can also configure settings specific to the Tempo data source. These options are described in the sections below.   
```

> There are two ways to configure the trace to logs feature. You can use simplified configuration with default query, or you can configure custom query where you can use a [template language]({{< relref "../../dashboards/variables/variable-syntax">}}) to interpolate variables from the trace or span.

Reviewer: ```suggestion
There are two ways to configure the trace to logs feature: 
* Use a simplified configuration with default query, or 
* Configure a custom query where you can use a [template language]({{< relref "../../dashboards/variables/variable-syntax">}}) to interpolate variables from the trace or span.
```

> #### To use a simple configuration:

Reviewer: ```suggestion
#### Use a simple configuration
```

1. Select the target data source.
1. Set start and end time shift. As the logs timestamps may not exactly match the timestamps of the spans in trace it may be necessary to search in larger or shifted time range to find the desired logs.
1. Select which tags to use in the logs query. The tags you configure must be present in the spans attributes or resources for a trace to logs span link to appear. You can optionally configure a new name for the tag. This is useful for example if the tag has dots in the name and the target data source does not allow using dots in labels. In that case you can for example remap `http.status` to `http_status`.
1. Optionally switch on the **Filter by trace ID** and/or **Filter by span ID** setting to further filter the logs if your logs consistently contain trace or span IDs.

> #### To use a custom query configuration:

Reviewer: ```suggestion
#### Configure a custom query
```

1. Switch on **Use custom query**.
1. Specify a custom query to be used to query the logs. You can use various variables to make that query relevant for current span. The link will only be shown only if all the variables are interpolated with non-empty values to prevent creating an invalid query.

#### Variables that can be used in a custom query

To use a variable you need to wrap it in `${}`. For example `${__span.name}`.

| Variable name          | Description                                                                                                                                                                                                                                                                                                                                                        |
| ---------------------- | ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------ |

> | **\_\_tags**           | This variable is special because it uses the tag mapping in from the UI to create a label matcher string in the specific data source syntax. It uses only tags that are present in the span so the link will still be created even if only one of those tags is present in the span. You can use this if not all the tags are required for the query to be useful. |

Reviewer: In the second sentence, it says: "...so the link will still be created even if only one of those tags is present in the span." I am not sure which (or what) link is created. Will it be clear to a reader what is meant? Maybe there is a way to reword this text so it's clearer, as well. 
```suggestion
| **\_\_tags**           | This variable uses the tag mapping in from the UI to create a label matcher string in the specific data source syntax. The variable only tags that are present in the span. The link is still created even if only one of those tags is present in the span. You can use this if all tags are not required for the query to be useful. |
```

| **\_\_span.traceId**   | The ID of the trace.                                                                                                                                                                                                                                                                                                                                               |
| **\_\_span.duration**  | The duration of the span.                                                                                                                                                                                                                                                                                                                                          |
| **\_\_span.name**      | Name of the span.                                                                                                                                                                                                                                                                                                                                                  |

> | **\_\_span.tags**      | Namespace for the tags in the span. To access a specific tag named "version" you would use `${__span.tags.version}`. In case the tag contains dot you have to access it as `${__span.tags["http.status"]}`.                                                                                                                                                        |

Reviewer: ```suggestion
| **\_\_span.tags**      | Namespace for the tags in the span. To access a specific tag named `version`, you would use `${__span.tags.version}`. In case the tag contains dot, you have to access it as `${__span.tags["http.status"]}`.                                                                                                                                                        |
```

The following table describes the ways in which you can configure your trace to logs settings:

| Setting name              | Description                                                                                                                                                                                                                                                                    |
| ------------------------- | ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------ |
| **Data source**           | Defines the target data source. You can select only Loki or Splunk \[logs\] data sources.                                                                                                                                                                                      |

> | **Span start time shift** | Shifts the start time for the logs query, based on the span's start time. You can use time units, such as `5s`, `1m`, `3h`. To extend the time to the past, use a negative value. Default: 0.                                                                                  |

Reviewer: ```suggestion
| **Span start time shift** | Shifts the start time for the logs query, based on the span's start time. You can use time units, such as `5s`, `1m`, `3h`. To extend the time to the past, use a negative value. Default: `0`.                                                                                  |
```

> | **Span end time shift**   | Shifts the end time for the logs query, based on the span's end time. You can use time units. Default: 0.                                                                                                                                                                      |

Reviewer: ```suggestion
| **Span end time shift**   | Shifts the end time for the logs query, based on the span's end time. You can use time units. Default: `0`.                                                                                                                                                                      |
```

> | **Tags**                  | Defines the the tags to use in the logs query. Default is `'cluster', 'hostname', 'namespace', 'pod'`. You can change the tag name for example to remove dots from the name if they are not allowed in the target data source. For example map `http.status` to `http_status`. |

Reviewer: ```suggestion
| **Tags**                  | Defines the the tags to use in the logs query. Default is `cluster`, `hostname`, `namespace`, `pod`. You can change the tag name for example to remove dots from the name if they are not allowed in the target data source. For example map `http.status` to `http_status`. |
```

  Interpolate tags using the `$__tags` keyword.
  For example, when you configure the query `requests_total{$__tags}`with the tags `k8s.pod=pod` and `cluster`, the result looks like `requests_total{pod="nginx-554b9", cluster="us-east-1"}`.

### Service Graph

The **Service Graph** section configures the [Service Graph](/docs/tempo/latest/grafana-agent/service-graphs/) feature.

Configure the **Data source** setting to define in which Prometheus instance the Service Graph data is stored.

> To use the Service Graph, refer to the [service graph documentation]({{< relref "#use-the-service-graph" >}}).

Reviewer: If Service Graph is a proper noun, then we should be consistent in capitalization. 
```suggestion
To use the Service Graph, refer to the [Service Graph documentation]({{< relref "#use-the-service-graph" >}}).
```
